Repository/ProductRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using EFCoreServices.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace EFCoreServices.Repository
+{
+    public class ProductRepository : IProductRepository
+    {
+        ProductDBContext db;
+        public ProductRepository(ProductDBContext _DbContext)
+        {
+            db = _DbContext;
+        }
+
+        public async Task<ProductDto> Add(Product prod)
+        {
+            
+                await db.AddAsync(prod);
+                await db.SaveChangesAsync();
+
+                var dto = new ProductDto()
```

Comment:
this needs to be done in a service class not repository 
repository should only deal with model